src/AutoRest.CSharp/Common/Output/Models/Types/ObjectTypeConstructor.cs

```diff
@@ -10,18 +12,18 @@ internal class ObjectTypeConstructor
     {
         public ObjectTypeConstructor(string name, string modifiers, Parameter[] parameters, ObjectPropertyInitializer[] initializers, ObjectTypeConstructor? baseConstructor = null)
         {
-            Signature = new MethodSignature(
+            Signature = new ConstructorSignature(
                 name,
                 $"Initializes a new instance of {name}",
                 modifiers,
                 parameters,
-                baseConstructor?.Signature);
+                new(isBase: true, baseConstructor?.Signature.Parameters ?? Array.Empty<Parameter>()));
```

Comment:
This is an example of a "temporary" code that has existed for several months. In the PR where `MethodSignature` type was introduced, it was just a bit easier to use the same type for initializer instead of a different one, and this place has slipped through the CR.
Initializer should've had its own type from the start.